Detections/MultipleDataSources/UserImpersonateByAAID.yaml

```diff
@@ -0,0 +1,67 @@
+id: 11c3d541-5fa5-49df-8218-d1c98584473b
+name: User impersonation by Identity Protection alerts
+description: |
+  'This detection focuses on identifying user-related events involving IAM roles, groups, user access, and password changes. It examines instances where the user's IP address matches and alerts generated by Identity Protection share the same IP address. The analysis occurs within a time window of 1 hour, helping to flag potential cases of user impersonation.'
+severity: Medium
+requiredDataConnectors:
+  - connectorId: AWS
+    dataTypes:
+      - AWSCloudTrail
+  - connectorId: AzureActiveDirectoryIdentityProtection
+    dataTypes:
+      - SecurityAlert (IPC)
+queryFrequency: 1d
+queryPeriod: 1d
+triggerOperator: gt
+triggerThreshold: 0
+tactics:
+  - PrivilegeEscalation
+relevantTechniques:
+  - T1134
+query: |
+ // Retrieve SecurityAlerts generated within the last day
+  SecurityAlert 
+  | where TimeGenerated >= ago(1d)
+  // Filter alerts for Azure Active Directory Identity Protection and High severity
+  | where ProductName has "Azure Active Directory Identity Protection"
+  | where AlertSeverity == "High"
+  // Extract IP address entities from the 'Entities' field
+  | extend ipAddress = extract(@'\b(?:\d{1,3}\.){3}\d{1,3}\b', 0, Entities)
+  // Filter out alerts without IP address entities
+  | where isnotempty(ipAddress)
+  // Summarize entities per unique combination of attributes
+  | summarize make_set(Entities)
+      by
+      AlertTime = TimeGenerated,
+      ipAddress,
+      AlertName,
+      ProductName,
+      AlertSeverity
+  // Perform an inner join with AWS CloudTrail events
+  | join kind=inner (
+      AWSCloudTrail
+      | where TimeGenerated >= ago(1d)
+      | where isempty(ErrorMessage)
+      //|where EventSource == "iam.amazonaws.com"

```

Comment:
Why is this commented out?

Reply:
we are expanding the coverage source, not limiting to IAM but it will also cover webmail and other Identity solutions

Reply:
if its generate any FP then we will narrow down the scope

Reply:
In that case we should either remove completely or add an extra comment to explain this to people using this template. Currently it's a bit confusing.

Reply:
I have removed